cpp/src/arrow/python/numpy_to_arrow.cc

```diff
@@ -372,45 +372,25 @@ Status StaticCastBuffer(const Buffer& input, const int64_t length, MemoryPool* p
   return Status::OK();
 }
 
-template <typename T>
-void CopyStridedBytewise(int8_t* input_data, int64_t length, int64_t stride,
-                         T* output_data) {
+void CopyStridedBytewise(const int8_t* input_data, const int64_t length,
+                         const int64_t itemsize, const int64_t stride,
+                         int8_t* output_data) {
```

Comment:
Pass input and output data as `uint8_t*` rather than `int8_t*`, it will avoid at least one of the `reinterpret_cast`s below.